src/main/kotlin/step2/calculator/ExpressionParser.kt

```diff
@@ -0,0 +1,41 @@
+package step2.calculator
+
+import step2.inspector.mustNot
+
+class ExpressionParser {
+    fun parse(expression: String) = expression
+        .splitToTextSymbolAndNumbers()
+        .convertToCalculations()
+
+    /**
+     * expression은 "2 + 3 * 4 / 2"와 같은 형태로 들어 온다.
+     * 이를 split하면 [2,+,3,*,4,/,2] 모양의 list를 얻을 수 있는데,
+     * list 맨 앞에 +를 더해줌으로써 '연산자->숫자'가 반복되는 규칙을 부여한다.
+     */
+    private fun String.splitToTextSymbolAndNumbers() = listOf(Operator.SYMBOL_PLUS) + this.split(DELIMITER)
+
+    @Throws(IllegalArgumentException::class, NumberFormatException::class)
+    private fun List<String>.convertToCalculations() = chunked(SIZE_OF_CALCULATION_UNIT) { symbolAndNumber ->
+        val textSymbol = symbolAndNumber.getOrNull(INDEX_SYMBOL)
+        val textNumber = symbolAndNumber.getOrNull(INDEX_NUMBER)
+        mustNot(IllegalArgumentException(INVALID_CALCULATION_VALUE_ERROR_MESSAGE)) {
+            textSymbol.isNullOrBlank() || textNumber.isNullOrBlank()
+        }
+
+        // 위에서 non-null을 확인했기 때문에 !!를 사용함.
+        Calculation(
+            operator = Operator.from(textSymbol!!),
+            number = textNumber!!.toDouble()
+        )
```

Comment:
`chunked`를 이용해 List가 2개라는 게 보장된다면 `List.first()`와 `List.last()`를 이용하고, `!!` 키워드를 제거하는 방향도 고려해볼 수 있었을 것 같네요~

```suggestion
    private fun List<String>.convertToCalculations() = chunked(SIZE_OF_CALCULATION_UNIT) { symbolAndNumber ->
        val textSymbol = symbolAndNumber.first()
        val textNumber = symbolAndNumber.last()
        mustNot(IllegalArgumentException(INVALID_CALCULATION_VALUE_ERROR_MESSAGE)) {
            textSymbol.isBlank() || textNumber.isBlank()
        }

        // 위에서 non-null을 확인했기 때문에 !!를 사용함.
        Calculation(
            operator = Operator.from(textSymbol),
            number = textNumber.toDouble()
        )
    }

```